Content.Server/Construction/ConstructionSystem.Graph.cs

```diff
@@ -298,9 +299,20 @@ public bool ChangeNode(EntityUid uid, EntityUid? userUid, string id, bool perfor
                 throw new Exception("Missing construction components");
             }
 
+            // Skip if the new entity's prototype is the same as the original, or the prototype is invalid
             if (newEntity == metaData.EntityPrototype?.ID || !_prototypeManager.HasIndex<EntityPrototype>(newEntity))
                 return null;
 
+            // Optional skip if the new entity's prototype is a parent of the original
```

Comment:
What is the purpose of this? Why is it needed? (Is it worth documenting in a comment here?)

Reply:
I've updated the technical details section of the PR to explain why its needed